rise-storage.html

```diff
@@ -473,35 +486,52 @@
     },
 
     /**
-     * Fires when an error is received from the Storage request.
+     * Retry the storage request one more time if it fails and attempt to load file from cache
      */
-    _handleStorageError: function(e, resp) {
-      var fireError = true;
+    _handleStorageErrorWithRetry: function() {
 
-      if (!this._isLoading && this._isCacheRunning && !this._fileThrottled) {
-        // storage request initiated from a refresh, Rise Cache is running, and no flag of previous file being throttled
+      var fireError = true;
 
-        if (this._isFile) {
-          fireError = false;
+      if ( !this._hasAttemptedSecondStorageRequest ) {
+        this._hasAttemptedSecondStorageRequest = true;
+        this._loadStorage();
+        fireError = false;
+      }
+      else {
+        if (!this._isLoading && this._isCacheRunning && !this._fileThrottled) {
+          // storage request initiated from a refresh, Rise Cache is running, and no flag of previous file being throttled
 
-          // proceed with getting file from Rise Cache
-          this._getFileFromCache();
-        }
-        else {
-          // folder files
-          if (this._folderFilesToRequest.length > 0) {
+          if (this._isFile) {
             fireError = false;
 
-            // reset the request list/count
-            this._folderFilesRequested = 0;
+            // proceed with getting file from Rise Cache
+            this._getFileFromCache();
+          }
+          else {
+            // folder files
+            if (this._folderFilesToRequest.length > 0) {
+              fireError = false;
+
+              // reset the request list/count
+              this._folderFilesRequested = 0;
 
-            // request the first folder file in the list to get cycle started
-            this._requestCacheFolderFile(this._folderFilesToRequest[0]);
+              // request the first folder file in the list to get cycle started
+              this._requestCacheFolderFile(this._folderFilesToRequest[0]);
+            }
           }
         }
-
       }
 
+      return fireError;
```

Comment:
@stulees it will set the fireError to false if it can make a request to rise cache or when making the retry request to storage. So the error event will not be fired unless it has already made the retry or it cannot get the file from rise cache. Does that make sense? cheers